src/Stack/MetricsCollector.php

```diff
@@ -0,0 +1,229 @@
+
+    private $registry;
+    private $metrics;
+    private $wpdb_stats;
+
+    public function __construct()
+    {
+        $this->metrics = new MetricsRegistry(
+            array(
+                'wp.requests'              => array('counter',   'Number of requests',                                 ['request_type']),
+                'wp.page_generation_time'  => array('histogram', 'Page generation time, in seconds',                   ['request_type']),
+                'wp.peak_memory'           => array('histogram', 'Peak memory per request, in bytes',                  ['request_type']),
+                'wpdb.query_time'          => array('histogram', 'Total MySQL query time per request, in seconds',     ['request_type']),
+                'wpdb.num_queries'         => array('histogram', 'Total number of MySQL queries per request',          ['request_type']),
+                'wpdb.num_slow_queries'    => array('histogram', 'Number of MySQL slow queries per request',           ['request_type']),
+                'wpdb.slow_query_treshold' => array('gauge',     'The treshold for counting slow queries, in seconds', []),
+                'woocommerce.orders'       => array('counter',   'Number of completed WooCommerce orders',             []),
+                'woocommerce.checkouts'    => array('counter',   'Number of started WooCommerce checkouts',            [])
+            )
+        );
+
+        $this->register_hooks();
+        $this->init_wpdb_stats();
+    }
+
+
+    public function init_wpdb_stats()
+    {
+        $this->wpdb_stats['slow_query_treshold'] = defined('SLOW_QUERY_THRESHOLD') ? SLOW_QUERY_THRESHOLD : 2000;
+        $this->wpdb_stats['query_time'] = 0;
+        $this->wpdb_stats['num_queries'] = 0;
+        $this->wpdb_stats['num_slow_queries'] = 0;
+    }
+
+    public function collect_request_metrics()
+    {
+        $request_type = $this::_get_request_type();
+        $display      = 0;
+        $precision    = 12;
+        $request_time = timer_stop($display, $precision);
+        $peak_memory  = memory_get_peak_usage();
+
+        $this->metrics->get_counter('wp.requests')->incBy(1, [$request_type]);
+        $this->metrics->get_histogram('wp.peak_memory')->observe($peak_memory, [$request_type]);
+        $this->metrics->get_histogram('wp.page_generation_time')->observe($request_time, [$request_type]);
+
+        if ($this::_do_collect_wpdb_metrics()) {
+            $this->metrics->get_histogram('wpdb.query_time')->observe($this->wpdb_stats['query_time'], [$request_type]);
+            $this->metrics->get_histogram('wpdb.num_queries')->observe($this->wpdb_stats['num_queries'], [$request_type]);
+            $this->metrics->get_histogram('wpdb.num_slow_queries')->observe($this->wpdb_stats['num_slow_queries'], [$request_type]);
+        }
+    }
+
+    public function register_endpoint()
+    {
+        $version   = '1';
```

Comment:
`$version` should be a `MetricsCollector ` attribute.

Reply:
Since this is the version number in the URL for the REST API endpoint (`/wp-json/stack/v1/metrics`), maybe I could make it a constant, something like `STACK_REST_API_VERSION`.